bin/add-server

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env node
+
+require('dotenv').config();
+const axios = require('axios');
+
+// step 1, check command line args
+const args = process.argv.slice(2); // first 2 are "node add-server"
+// console.log(args);
+
+const newServer = {};
+
+for (let i = 0 ; i < args.length ; i++) {
+  const argKey = args[i];
+  const argValue = args[++i];
+
+  switch (argKey) {
+    case '-e':
+    case '--endpoint':
+      newServer['endpoint'] = argValue;
+      break;
+
+    case '-t':
+    case '--type':
+      newServer['type'] = argValue;
+      break;
+
+    case '-c':
+    case '--clientId':
+      newServer['clientId'] = argValue;
+      break;
+
+    case '-n':
+    case '--name':
+      newServer['name'] = argValue;
+      break;
+
+    default:
+      console.error(`Unknown command line option: ${argKey}`);
+      console.error(`Full args: ${args}`);
+      process.exit(1);
+  }
+}
+
+if (!newServer['endpoint']) {
+  console.error("Required argument -e|--endpoint missing!");
+  process.exit(1);
+}
+
+if (!newServer['name']) {
+  // if a name wasn't provided, use the endpoint URL as the name
+  newServer['name'] = newServer['endpoint'];
+}
+
+const baseURL = process.env.BASE_URL;
+
+// see if the server is already running
+// by hitting the base url.
+// no connection = it's not running, add to DB directly
+// connection succeeds = it's running, use the API
+
+axios.get(baseURL).then(callApi).catch(addToDb);
+
+function addToDb() {
+  console.log('App not running. Adding server to db directly.');
+
+  const servers = require('../src/storage/servers');
+  // TODO: this doesn't actually work. need to async wait for it to init
+  // and even then it seems to hang for no reason
+  servers.addServer(newServer);
+  console.log('Added successfully.');
```

Comment:
So you need to wait for to initialize but then it hangs because the db is still running. I couldn't find a way to have loki shut down so using `process.exit(0)` did the trick... but you also have to wait for `servers.addServer()` to complete before calling `exit`. This is not a pretty solution but it worked for me 
```
const servers = require('../src/storage/servers');
  setTimeout(() => {
    servers.addServer(newServer);
    setTimeout(() => {
      console.log('Added successfully.');
      process.exit(0);
    }, 10000);
  }, 10000);
```

Reply:
Yeah this is where that idea I had of refactoring the DB post-init stuff would be useful. Not sure if I want to do something like this first, then refactor later, or just go for it and refactor things here

Reply:
I'm good to refactor it later. Created [MEDMORPH-77](https://jira.mitre.org/browse/MEDMORPH-77).

Reply:
Too late I did it here. It's only like 10 lines.